docs/guide/apm-data-model.asciidoc

```diff
@@ -144,4 +157,101 @@ For a full list of tracked metrics, see the relevant agent documentation:
+Ruby: {apm-ruby-ref}/api.html#api-agent-set-tag[`set_tag`]
+Rum: {apm-rum-ref}/api.html#apm-add-tags[`addTags`]
+
+|Custom
+|No
+v|<<transactions>>
+<<errors>>
+v|Go: {apm-go-ref}/api.html#slug-here[`???`]
+Java: {apm-java-ref}/public-api.html#slug-here[`???`]
+Node.js: {apm-node-ref}/agent-api.html#apm-set-custom-context[`setCustomContext`]
+Python: {apm-py-ref}/api.html#api-set-custom-context[`set_custom_context`]
+Ruby: {apm-ruby-ref}/api.html#api-agent-set-custom-context[`set_custom_context`]
+Rum: {apm-rum-ref}/api.html#apm-set-custom-context[`setCustomContext`]
+
+|User
+|Yes
+v|<<transactions>>
+<<errors>>
+v|Go: {apm-go-ref}/api.html#context-set-username[`SetUsername`] \| {apm-go-ref}/api.html#context-set-id[`SetUserID`] \| {apm-go-ref}/api.html#context-set-email[`SetUserEmail`]
+Java: {apm-java-ref}/public-api.html#api-transaction-set-user[`setUser`]
+Node.js: {apm-node-ref}/agent-api.html#apm-set-user-context[`setUserContext`]
+Python: {apm-py-ref}api.html#api-set-user-context[`set_user_context`]
+Ruby: {apm-ruby-ref}/api.html#api-agent-set-user[`set_user`]
+Rum: {apm-rum-ref}/api.html#apm-set-user-context[`setUserContext`]
+
+|=======================================================================
+
+// GO CUSTOM - not quite sure where to link
+// JAVA CUSTOM - looks like Java doesn't have a custom context option
+
+[[labels-fields]]
+==== Labels
+
+Labels are used to add *indexed* information to transactions, spans, and errors.
+Indexed means the data is searchable and aggregatable in Elasticsearch.
+Multiple labels can be defined with different key-value pairs.
+
+* Elasticsearch type: {ref}/object.html[object]
+* Elasticsearch field: `labels` (previously `context.tags`)
+
+Label values can be a string, boolean, or number.
+Note that some agents only support string values at this time.
+Because labels are stored in the same place in Elasticsearch, each value must have the same datatype.
+Multiple datatypes per key will throw an exception, e.g. `{foo: bar}` and `{foo: 42}`
+
+IMPORTANT: Avoid defining too many user-specified labels.
+Defining too many unique fields in an index is a condition that can lead to a
+{ref}/mapping.html#mapping-limit-settings[mapping explosion].
+
+// elastic/apm-server#828
+// https://github.com/elastic/apm-server/pull/1712
+
+[[custom-fields]]
+==== Custom context
+
+Custom context is used to add *non-indexed*,
+custom contextual information to transactions and errors.
+Non-indexed means the data is not searchable or aggregatable in Elasticsearch,
+and you cannot build dashboards on top of the data.
+However, non-indexed information is useful for other reasons,
+like providing contextual information to help you quickly debug performance issues or errors.
+
+* Elasticsearch type: {ref}/object.html[object]
+* Elasticsearch fields: `transaction.custom` | `error.custom`
+
+IMPORTANT: Setting a circular object, large object, or a non JSON serializable object can lead to errors.
+
```

Comment:
Maybe add a note that mapping explosions are not a concern here because it does not add fields to the mapping? At least I think that this is the case.

Reply:
Added above